src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -243,13 +243,35 @@ impl<'a, 'tcx> Qualifier<'a, 'tcx, 'tcx> {
             return;
         }
 
+        if self.tcx.features().const_let {
+            let mut dest = dest;
+            let index = loop {
+                match dest {
+                    // with `const_let` active, we treat all locals equal
+                    Place::Local(index) => break *index,
+                    // projections are transparent for assignments
+                    // we qualify the entire destination at once, even if just a field would have
+                    // stricter qualification
```

Comment:
This doesn't seem correct - if you assign something with less qualifications to a field of a value with more, you turn those bits off.